sdk/core/azure-core/src/logging/logging.cpp

```diff
@@ -7,72 +7,41 @@
 #include <mutex>
 
 using namespace Azure::Core::Logging;
-using namespace Azure::Core::Logging::Details;
-
-class Azure::Core::Logging::Details::LogClassificationsPrivate {
-public:
-  static LogClassifications const LogClassificationsConstant(bool all)
-  {
-    return LogClassifications(all);
-  }
-
-  static bool IsClassificationEnabled(LogClassifications const& cls, LogClassification const& c)
-  {
-    return cls.m_all || (cls.m_classifications.find(c) != cls.m_classifications.end());
-  }
-};
+using namespace Azure::Core::Logging::Internal;
 
 namespace {
 std::mutex g_loggerMutex;
 LogListener g_logListener(nullptr);
+LogLevel g_logLevel = LogLevel::Verbose;
 
-LogClassifications g_logClassifications(
-    LogClassificationsPrivate::LogClassificationsConstant(true));
-
-LogListener GetLogListener(LogClassification const& classification)
+LogListener GetLogListener(LogLevel level)
 {
-  // lock listener and classifications
   std::lock_guard<std::mutex> loggerLock(g_loggerMutex);
-
-  return (!g_logListener // if no logger is set
-          || LogClassificationsPrivate::IsClassificationEnabled(
-              g_logClassifications, classification))
-      // or if classification is enabled
-      ? g_logListener // return actual listener (may be null)
-      : LogListener(nullptr) // return null listener
-      ;
+  return (g_logListener && g_logLevel <= level) ? g_logListener : LogListener(nullptr);
```

Comment:
If you use atomic, you'd copy g_log_listener into a local variable and use the local variable twice on the return line.